src/vscode-bicep/src/language/protocol.ts

```diff
@@ -80,6 +83,43 @@ export interface BicepDeploymentWaitForCompletionParams {
   documentPath: string;
 }
 
+export interface BicepDeploymentParameter {
+  name: string;
+  value?: string | undefined;
+  isMissingParam: boolean;
+  isExpression: boolean;
+  isSecure: boolean;
+  parameterType: ParameterType | undefined;
+}
+
+export interface BicepDeploymentParametersResponse {
+  deploymentParameters: BicepDeploymentParameter[];
+  parametersFileName: string;
+  errorMessage?: string;
+}
+
+export interface BicepUpdatedDeploymentParameter {
+  name: string;
+  value: string;
+  isSecure: boolean;
+  parameterType: ParameterType | undefined;
+}
+
+export enum ParametersFileUpdateOption {
+  Create = 1,
+  None = 2,
+  Overwrite = 3,
```

Comment:
I don't understand when Overwrite is used

Reply:
Added comments here:
https://github.com/Azure/bicep/commit/f872da8a7fb96fa1169b2db62eed7038c66c9fa7

Overwrite is used in this scenario:
- If the user did not provide a parameters file to be used in deployment,
- and chose to create one at the end of the flow
- and file with name <bicep_file_name>.parameters.json already exists in the same folder as bicep file